### PR TITLE
refactor: `CreateANewTeamForm`

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR is stacked on #11169

When using SCIM for groups we need to map the group name to the team in the Cal org. For better flow we should allow IT admins to create the team under an org on this page. That means refactoring `CreateANewTeamForm`.

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Create a new team as normal

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by joeauyeung** [comment]
Current dependencies on/for this PR:
* **#13818** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13818?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `feat/scim-dsync`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13818?utm_source=stack-comment).

File: apps/web/pages/settings/teams/new/index.tsx
Comment (joeauyeung) [inline apps/web/pages/settings/teams/new/index.tsx:41]: Should we track telemetry for all created teams or just from the normal flow?

File: packages/features/ee/teams/components/CreateANewTeamForm.tsx
Comment (joeauyeung) [inline packages/features/ee/teams/components/CreateANewTeamForm.tsx:1]: The idea of reusing this form is not to repeat the error handling logic.

File: apps/web/pages/settings/teams/new/index.tsx
Comment (CarinaWolli) [inline apps/web/pages/settings/teams/new/index.tsx:54]: Do we need to pass the slug here? 
`slug={parsedQuery.success ? parsedQuery.data.slug : ""}`

**Review by CarinaWolli — APPROVED** [review]
Code looks good 🙏

**Comment by joeauyeung** [comment]
Let's not merge until #13850 is merged into main

**Comment by PeerRich** [comment]
> Let's not merge until https://github.com/calcom/cal.com/pull/13850 is merged into main

oops

**Pull request merged**
